poly-commitment/src/srs.rs

```diff
@@ -1,19 +1,23 @@
 //! This module implements the Marlin structured reference string primitive
 
 use crate::commitment::CommitmentCurve;
-use ark_ec::{AffineCurve, ProjectiveCurve};
+use ark_ec::{
+    short_weierstrass_jacobian::GroupAffine, AffineCurve, ModelParameters, ProjectiveCurve,
+};
 use ark_ff::{BigInteger, PrimeField};
 use ark_poly::{EvaluationDomain, Radix2EvaluationDomain as D};
 use array_init::array_init;
 use blake2::{Blake2b512, Digest};
 use groupmap::GroupMap;
+use mina_curves::pasta::{pallas::PallasParameters, vesta::VestaParameters};
+use oracle::poseidon::ArithmeticSpongeParams;
 use serde::{Deserialize, Serialize};
 use serde_with::serde_as;
 use std::collections::HashMap;
 
 #[serde_as]
 #[derive(Debug, Clone, Default, Serialize, Deserialize)]
-pub struct SRS<G: CommitmentCurve> {
+pub struct SRS<G: CommitmentCurve + KimchiCurve> {
```

Comment:
does the SRS really need a `KimchiCurve` here? Can we get away with just a `CommitmentCurve`? (which would be useful for non-recursion applications)

Reply:
removed